src/utility/ATT.cpp

```diff
@@ -683,7 +683,8 @@ void ATTClass::findInfoReq(uint16_t connectionHandle, uint16_t mtu, uint8_t dlen
     BLELocalAttribute* attribute = GATT.attribute(i);
     uint16_t handle = (i + 1);
     bool isValueHandle = (attribute->type() == BLETypeCharacteristic) && (((BLELocalCharacteristic*)attribute)->valueHandle() == handle);
-    int uuidLen = isValueHandle ? 2 : attribute->uuidLength();
+    bool isDescriptor = attribute->type() == BLETypeDescriptor;
+    int uuidLen = isValueHandle ? 2 : (isDescriptor ? attribute->uuidLength() : BLE_ATTRIBUTE_TYPE_SIZE);
```

Comment:
I don't think this is correct. If it's a value handle, it should use the attribute's UUID length right?

If we align it with the `if` logic on 703, value and descriptor handles memcpy the UUID data, so the uuidLen should reflect as such. Otherwise use the type size of 2
```C
int uuidLen = (isValueHandle || isDescriptor) ? attribute->uuidLength() : BLE_ATTRIBUTE_TYPE_SIZE;
// also can update the if on 703 to match
if (isValueHandle || isDescriptor) {
    // add the UUID
} else {
    // add the type
}
```

Reply:
Yes you're definitely right!
I'm going to update the PR